### PR TITLE
Add CancelAll API call

### PR DESCRIPTION
**Pull request by lt**
This adds the ability to cancel all open orders in a single call.

The public sandbox seems to be down at the moment, so even though I have written a test, I can't run it.

### CODE REVIEW

File: order.go [hunk 1]
```diff
@@ -52,6 +52,12 @@ func (c *Client) CancelOrder(id string) error {
 	return err
 }
 
+func (c *Client) CancelAll() ([]string, error) {
```
Comment (preichenberger) [inline order.go]: Can we rename this to CancelAllOrders?

Also can we have this support the optional product_id query parameter? You can do something like with ListOrdersParams and then: func (c *Client) CancelAllOrders(p ...CancelAllOrdersParams) ([]string, error)

File: order_test.go [hunk 2]
```diff
@@ -132,3 +132,19 @@ func TestListOrders(t *testing.T) {
 		}
 	}
 }
+
+func TestCancelAll(t *testing.T) {
+	client := NewTestClient()
+
```
Comment (preichenberger) [inline order_test.go]: Won't this always fail if it's run without any outstanding orders?

File: order.go (hunk 1 above)
Comment (lt) [reply order.go]: Absolutely. Would you prefer it to be an "optional" parameter, or two distinct methods; `CancelAllOrders` and `CancelAllOrdersForProduct` ?

File: order_test.go (hunk 2 above)
Comment (lt) [reply order_test.go]: I'm not sure. :) 

Not very experienced with Go, and with the public sandbox being down I couldn't actually run the test suite. Seems to still be down right now.

I'll add some glue to ensure it always has an order to cancel anyway.

File: order.go (hunk 1 above)
Comment (preichenberger) [reply order.go]: "optional" parameter

File: order_test.go (hunk 2 above)
Comment (preichenberger) [reply order_test.go]: I think if you make an order with a really high size, that should be sufficient

**Comment by preichenberger** [comment]
Hi, there were a number of issues. I went ahead and put it a commit and gave you a shout out!

https://github.com/preichenberger/go-gdax/commit/a9c11d54432fa151e34d836d69770d7ff3e1d6ef

**Pull request closed without merging**